client/lib/interval/index.js

```diff
@@ -0,0 +1,64 @@
+import React, { PropTypes } from 'react';
+
+/**
+ * Calls a given function on a given interval
+ */
+export default React.createClass( {
+	displayName: 'Interval',
+
+	propTypes: {
+		onTick: PropTypes.func.isRequired,
+		milliseconds: PropTypes.number,
+		leading: PropTypes.bool,
+		pauseWhenHidden: PropTypes.bool,
+		children: PropTypes.element
+	},
+
+	getDefaultProps: () => ( {
+		milliseconds: 30000,
+		leading: true,
+		pauseWhenHidden: true
+	} ),
+
+	getInitialState: () => ( {
+		timer: null
+	} ),
+
+	componentDidMount() {
+		this.start()
+	},
+
+	componentWillUnmount() {
+		this.stop()
+	},
+
+	componentDidUpdate() {
+		this.start();
+	},
+
+	run() {
+		clearTimeout( this.state.timer );
+
+		if ( document.hidden && this.props.pauseWhenHidden ) {
+			return this.setState( { timer: null } );
+		}
+
+		this.setState( { timer: setTimeout( this.run, this.props.milliseconds ) } );
+
+		this.props.onTick();
+	},
+
+	start() {
+		return ! this.state.timer && this.setState( {
```

Comment:
I haven't intended to actually return anything here. This is probably just confusing because I was trying to keep this as slim as possible. The reason I wrote it this way was so that we would abort the function if a timer were already set, meaning the call to `onTick()` is already looming. Otherwise, we accomplish the goal of scheduling _and_ setting the timer handle in one run with `setState()`